packages/app-store/office365calendar/lib/CalendarService.ts

```diff
@@ -246,9 +257,10 @@ export default class Office365CalendarService implements Calendar {
 
     return {
       getToken: () =>
+        refreshTokenResponseSchema.safeParse(o365AuthCredentials).success &&
         !isExpired(o365AuthCredentials.expires_in)
```

Comment:
Refresh the tokens if the keys object is not what we expected or the access token is expired